addons/account/models/account_lock_exception.py

```diff
@@ -0,0 +1,190 @@
+from odoo import _, api, fields, models, Command
+from odoo.tools import create_index
+from odoo.tools.misc import format_datetime
+from odoo.exceptions import UserError
+
+from odoo.addons.account.models.company import SOFT_LOCK_DATE_FIELDS
+
+
+class AccountLockException(models.Model):
+    _name = "account.lock_exception"
+    _description = "Account Lock Exception"
+
+    active = fields.Boolean('Active', default=True)
+    state = fields.Selection(
+        selection=[
+            ('active', 'Active'),
+            ('revoked', 'Revoked'),
+            ('expired', 'Expired'),
+        ],
+        string="State",
+        compute='_compute_state',
+        search='_search_state'
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.company,
+    )
+    # An exception w/o user_id is an exception for everyone
+    user_id = fields.Many2one(
+        'res.users',
+        string='User',
+        default=lambda self: self.env.user,
+    )
+    reason = fields.Char(
+        string='Reason',
+    )
+    # An exception without `end_datetime` is valid forever
+    end_datetime = fields.Datetime(
+        'End Date',
+    )
+
+    # Lock date fields; c.f. res.company
+    # An unset lock date field means the exception does not change this field.
+    # (It is not possible to remove a lock date completely).
+    fiscalyear_lock_date = fields.Date(
+        string="Global Lock Date",
+        help="The date the Global Lock Date is set to by this exception. If no date is set the lock date is not changed.",
+    )
+    tax_lock_date = fields.Date(
+        string="Tax Return Lock Date",
+        help="The date the Tax Lock Date is set to by this exception. If no date is set the lock date is not changed.",
+    )
+    sale_lock_date = fields.Date(
+        string='Lock Sales',
+        help="The date the Sale Lock Date is set to by this exception. If no date is set the lock date is not changed.",
+    )
+    purchase_lock_date = fields.Date(
+        string='Lock Purchases',
+        help="The date the Purchase Lock Date is set to by this exception. If no date is set the lock date is not changed.",
+    )
```

Comment:
Shouldn't these fields be set in `sale` and `purchase` modules instead? 🤔

Reply:
The Sale and Purchase Lock Dates lock journals with type `sale` or `purchase`. So they are about (`account`) moves.